### DISCUSSION TITLE
App storage shared sync with user defaults

### DISCUSSION THREAD

**Discussion by jiledaa** [pointfreeco/swift-composable-architecture#3314] [Ideas]
Hello guys. Can be somehow ```@Shared(.appStorage("someKey")) var``` connected to existing user defaults property via the key? I have notification extension executing even when the app is not alive, where I dont want to have the Composable architecture dependency but I have access to user defaults there. When I change user defaults property there, it would be great if the value is synced with app storage shared in my app so I could directly use the value when the app runs.

**Comment by mbrandonw** [comment]
> Can be somehow `@Shared(.appStorage("someKey")) var` connected to existing user defaults property via the key?

Hi @jiledaa, that is already the case. Can you describe your exact use case where this is not happening?

**Reply by jiledaa** [reply]
Hi, sure.  Have this registered in userdefaults:

```
    public enum SharedUserDefaultsKey: String {
        case missedCallsCount
    }

    var missedCallsCount: Int {
        get { self.integer(forKey: Constants.SharedUserDefaultsKey.missedCallsCount.rawValue) }
        set { self.set(newValue, forKey: Constants.SharedUserDefaultsKey.missedCallsCount.rawValue) }
    }
```

Have function in My2NNotificationExtensionManager, executing when missed notification is received (even if app is killed):
```
    @MainActor
    private func handleMissedCall(mutableContent: UNMutableNotificationContent) async {
            userDefaults?.missedCallsCount += 1
            mutableContent.badge = (userDefaults?.missedCallsCount ?? 0) as NSNumber
    }
```

Then registered app storage value like:
```
extension PersistenceReaderKey where Self == PersistenceKeyDefault<AppStorageKey<Int>> {
    static var missedCallsCount: Self {
        PersistenceKeyDefault(.appStorage(SharedUserDefaultsKey.missedCallsCount.rawValue), 0)
    }
}
```

Then just simply have @Shared(.missedCallsCount) var missedCallsCount in my feature.
I was hoping that once the missed call notification comes,  I can see that user defaults incrementation in my view via shared missedCallsCount, but it is still 0 despite the user defaults value is 1.
